src/Models/MetadataRelationsTrait.php

```diff
@@ -97,42 +50,43 @@ protected function getCodeForMethod(\ReflectionMethod $method) : string
         }
 
         $code = trim(preg_replace('/\s\s+/', '', $code));
-        if (false === stripos($code, 'function')) {
-            $msg = 'Function definition must have keyword \'function\'';
-            throw new InvalidOperationException($msg);
-        }
         $begin = strpos($code, 'function(');
         $code = substr($code, $begin, strrpos($code, '}') - $begin + 1);
         $lastCode = $code[strlen($code) - 1];
-        if ('}' != $lastCode) {
```

Comment:
This if check was dropped because we could not get a valid ReflectionMethod from invalid code.